File: Backend/models/Exercise.js
const mongoose = require('mongoose');

const ExerciseSchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: { type: String },
  duration: { type: Number, default: 0 }, // in minutes
  target: { type: String, required: true }, // e.g., 'Back', 'Legs'
  difficulty: { type: String, enum: ['Easy','Medium','Hard'], default: 'Easy' },
  day: { type: String, required: true } // e.g., 'Monday'
}, { timestamps: true });

module.exports = mongoose.model('Exercise', ExerciseSchema);

File: streakfit_app_fr/lib/models/exercise.dart
class Exercise {
  final String id;
  final String title;
  final String description;
  final int duration;
  final String target;
  final String day;
  Exercise({required this.id, required this.title, required this.description, required this.duration, required this.target, required this.day});
  static int _parseDuration(dynamic value) {
    if (value is int) return value;
    if (value is double) return value.round();
    return int.tryParse(value?.toString() ?? '') ?? 0;
  }
  factory Exercise.fromJson(Map<String,dynamic> j) => Exercise(
    id: j['_id'] ?? j['id'] ?? '',
    title: j['title'] ?? '',
    description: j['description'] ?? '',
    duration: _parseDuration(j['duration']),
    target: j['target'] ?? '',
    day: j['day'] ?? '',
  );
}
